EWL-4424 Adds link to body copy

designerneil opened the pull request:
Adds class to underline all links inside of p tags

<!-- NOTE: Please just put "N/A" for any section below that isn't applicable to the work you've done, do not omit entirely. -->

## Ticket(s)

Please do not submit a Pull Request without a relevant JIRA ticket or Github issue! If you are creating a new pattern or feature, create an issue describing the need for this feature in Github **first** and then link to it below.

**Jira Ticket**

- [EWL-4424: SG2 | Create News Article Template](https://issues.ama-assn.org/browse/EWL-4424)


## Description

Adds link copy to body as requested.

## To Test

- [ ] http://localhost:3000/?p=pages-news
- [ ] Observe the link in the body on the first line
- [ ] Observe the margins between paragraphs
- [ ] Observe the margin on left of the article which matches the inline bio margin text

## Visual Regressions

none


## Relevant Screenshots/GIFs

<img width="1226" alt="screen shot 2018-01-30 at 10 46 17 am" src="https://user-images.githubusercontent.com/2271747/35578851-d30e6e70-05aa-11e8-824d-16e581818127.png">

## Remaining Tasks

Remaining tasks?


## Additional Notes

Anything more to add? Good things to call out here are:
- are there any PRs in this or other repositories that relate to or affect this PR?
- are there any caveats or oddities testers should know about when testing this PR?
- are there any problems you ran into during your work that you'd like to call out?

csj1012 committed:
[EWL-4388] added pattern files

csj1012 committed:
[EWL-4388] added pattern files

csj1012 committed:
[EWL-4388] added pattern files

csj1012 committed:
[EWL-4388] add pattern to the Topic page pattern and start populating .json data

csj1012 committed:
[EWL-4388] add pattern to the Topic page pattern and start populating .json data

csj1012 committed:
[EWL-4388] add pattern to the Topic page pattern and start populating .json data

designerneil committed:
Adds article small variant to lead news

designerneil committed:
Adds article small variant to lead news

designerneil committed:
Merge branch 'feature/EWL-4388-migrate-lead-news-block' of github.com:AmericanMedicalAssociation/ama-style-guide-2 into feature/EWL-4388-migrate-lead-news-block

designerneil committed:
Adds lead news markdown

designerneil committed:
Adds lead news markdown

designerneil committed:
removes old colors and fonts
updates all files that used them

designerneil committed:
removes old colors and fonts
updates all files that used them

designerneil committed:
Adds new reds and greens

designerneil committed:
Adds new reds and greens

designerneil committed:
Merge branch 'feature/EWL-4388-migrate-lead-news-block' of github.com:AmericanMedicalAssociation/ama-style-guide-2 into feature/EWL-4388-migrate-lead-news-block

designerneil committed:
Fixes lead news per PR
- Adds related content article stub
- modifies title only
- modifies metadata to use a parent container
- changes some of the conditionals for article stub

designerneil committed:
Fixes type sizes
take away unnecessary parent selector in json

designerneil committed:
removes paragraph from json

designerneil committed:
Adds link to body copy
Adds class to underline all links inside of p tags

designerneil committed:
Adds link to body copy
Adds class to underline all links inside of p tags

designerneil committed:
Adds link to body copy
Adds class to underline all links inside of p tags

designerneil committed:
Adds link to body copy
Adds class to underline all links inside of p tags

designerneil committed:
Adds link to body copy
Adds class to underline all links inside of p tags

designerneil committed:
Adds link to body copy
Adds class to underline all links inside of p tags

designerneil committed:
Adds link to body copy
Adds class to underline all links inside of p tags

designerneil committed:
Adds link to body copy
Adds class to underline all links inside of p tags

designerneil committed:
Adds link to body copy
Adds class to underline all links inside of p tags

designerneil committed:
Adds link to body copy
Adds class to underline all links inside of p tags

designerneil committed:
Adds link to body copy
Adds class to underline all links inside of p tags

designerneil committed:
Adds link to body copy
Adds class to underline all links inside of p tags

designerneil committed:
Adds link to body copy
Adds class to underline all links inside of p tags

designerneil committed:
Adds link to body copy
Adds class to underline all links inside of p tags

designerneil committed:
Adds link to body copy
Adds class to underline all links inside of p tags

designerneil committed:
Adds link to body copy
Adds class to underline all links inside of p tags

designerneil committed:
Adds link to body copy
Adds class to underline all links inside of p tags

designerneil committed:
Adds link to body copy
Adds class to underline all links inside of p tags

designerneil committed:
Adds link to body copy
Adds class to underline all links inside of p tags

designerneil committed:
Adds link to body copy
Adds class to underline all links inside of p tags

designerneil committed:
Adds link to body copy
Adds class to underline all links inside of p tags

designerneil committed:
Adds link to body copy
Adds class to underline all links inside of p tags

designerneil committed:
Adds link to body copy
Adds class to underline all links inside of p tags

designerneil committed:
Adds link to body copy
Adds class to underline all links inside of p tags

designerneil committed:
Adds link to body copy
Adds class to underline all links inside of p tags

designerneil committed:
Adds link to body copy
Adds class to underline all links inside of p tags

designerneil committed:
Fixes type size for teaser

designerneil committed:
Adds social share buttons

designerneil committed:
Adds social share buttons

designerneil committed:
Adds social share buttons

designerneil committed:
Adds social share buttons

designerneil committed:
changes date block to use h3

designerneil committed:
Adjust bio image to have a max of 80px

designerneil committed:
tags now have border on top and button
tags uses flexbox instead of inline elements
margin and padding have been adjusted to give the tags more air

designerneil committed:
Adds learn more to tools block

Commit:
Merge pull request #70 from AmericanMedicalAssociation/bugfix/EWL-4419-date-atom

EWL-4419 changes date atom to use h3

Commit:
fix empty line

Commit:
Merge pull request #73 from AmericanMedicalAssociation/bugfix/EWL-4430-tools-list-footer

EWL-4430 Adds learn more to tools block

Commit:
Merge pull request #72 from AmericanMedicalAssociation/bugfix/EWL-4423-tag-atom

EWL-4423 Adjust tags styling and spacing

Commit:
Merge pull request #71 from AmericanMedicalAssociation/bugfix/EWL-4420-inline-bio

EWL-4420 Adjust bio image to have a max of 80px

Commit:
Merge pull request #68 from AmericanMedicalAssociation/bugfix/EWL-4433-Column-teaser

EWL-4433 Fixes type size for the column teaser to be smaller

labbydev reviewed (changes requested):
@designerneil per the comments in the ticket this updates should also include adding back the margins between the paragraphs. Please add that as well and resolved the merge conflicts.

designerneil committed:
tools clean and integration with D8

designerneil committed:
updates per PR suggestion

designerneil committed:
Address concerns made in PR
removes unused font mixin file

designerneil committed:
Merge branch 'bugfix/EWL-4424-Create-News-Article-Template' of github.com:AmericanMedicalAssociation/ama-style-guide-2 into bugfix/EWL-4424-Create-News-Article-Template

designerneil committed:
Adds margins in between paragraphs

designerneil committed:
Adds margins in between paragraphs

designerneil committed:
Adds margins in between paragraphs

designerneil committed:
Adds margins in between paragraphs

designerneil committed:
Adds margins in between paragraphs

designerneil committed:
Adds margins in between paragraphs

designerneil committed:
Adds margins in between paragraphs

designerneil committed:
Adds margins in between paragraphs

designerneil committed:
Adds margins in between paragraphs

designerneil committed:
Adds margins in between paragraphs

designerneil committed:
Adds margins in between paragraphs

designerneil committed:
Adds margins in between paragraphs

designerneil committed:
Adds margins in between paragraphs

designerneil committed:
Adds margins in between paragraphs

designerneil committed:
Adds margins in between paragraphs

designerneil committed:
Adds margins in between paragraphs

designerneil committed:
Adds margins in between paragraphs

designerneil committed:
Adds margins in between paragraphs

designerneil committed:
Adds margins in between paragraphs

designerneil committed:
Adds margins in between paragraphs

designerneil committed:
Adds margins in between paragraphs

designerneil committed:
Adds margins in between paragraphs

designerneil committed:
Adds margins in between paragraphs

designerneil committed:
Adds margins in between paragraphs

designerneil committed:
Merge branch 'bugfix/EWL-4418-social-share' of github.com:AmericanMedicalAssociation/ama-style-guide-2 into bugfix/EWL-4418-social-share

designerneil committed:
Adds empty line

designerneil committed:
Adds empty line

designerneil committed:
adds learn more tools footer

Commit:
Add empty line

Commit:
Merge pull request #74 from AmericanMedicalAssociation/feature/EWL-4165-tools

tools clean and integration with D8

Commit:
Add empty line

Commit:
Merge pull request #66 from AmericanMedicalAssociation/bugfix/EWL-4432-four-up-teaser

Fixes type sizes for Article Stub and Four Up Teaser

labbydev reviewed (changes requested):
@designerneil this includes the margins on the paragraphs however, I think what UX was looking for was multiple shorter paragraphs to match the wireframe. Please review the paragraph spacing here (https://sketch.cloud/s/L0dLQ/p/news-article/article-desktop) and try to align the prototype more closely to the look of the paragraphs there.

designerneil committed:
Adds margins between paragraphs to match mockup
Adds margin to the left of the article to match mockup

designerneil committed:
Adds margins between paragraphs to match mockup
Adds margin to the left of the article to match mockup

designerneil committed:
Adds margins between paragraphs to match mockup
Adds margin to the left of the article to match mockup

designerneil committed:
Adds margins between paragraphs to match mockup
Adds margin to the left of the article to match mockup

designerneil committed:
Adds margins between paragraphs to match mockup
Adds margin to the left of the article to match mockup

designerneil committed:
Adds margins between paragraphs to match mockup
Adds margin to the left of the article to match mockup

designerneil committed:
Adds margins between paragraphs to match mockup
Adds margin to the left of the article to match mockup

designerneil committed:
Adds margins between paragraphs to match mockup
Adds margin to the left of the article to match mockup

designerneil committed:
Adds margins between paragraphs to match mockup
Adds margin to the left of the article to match mockup

designerneil committed:
Adds margins between paragraphs to match mockup
Adds margin to the left of the article to match mockup

designerneil committed:
Adds margins between paragraphs to match mockup
Adds margin to the left of the article to match mockup

designerneil committed:
Adds margins between paragraphs to match mockup
Adds margin to the left of the article to match mockup

designerneil committed:
Adds margins between paragraphs to match mockup
Adds margin to the left of the article to match mockup

designerneil committed:
Adds margins between paragraphs to match mockup
Adds margin to the left of the article to match mockup

designerneil committed:
Adds margins between paragraphs to match mockup
Adds margin to the left of the article to match mockup

designerneil committed:
Adds margins between paragraphs to match mockup
Adds margin to the left of the article to match mockup

designerneil committed:
Adds margins between paragraphs to match mockup
Adds margin to the left of the article to match mockup

designerneil committed:
Adds margins between paragraphs to match mockup
Adds margin to the left of the article to match mockup

designerneil committed:
Adds margins between paragraphs to match mockup
Adds margin to the left of the article to match mockup

designerneil committed:
Adds margins between paragraphs to match mockup
Adds margin to the left of the article to match mockup

designerneil committed:
Adds margins between paragraphs to match mockup
Adds margin to the left of the article to match mockup

designerneil committed:
Adds margins between paragraphs to match mockup
Adds margin to the left of the article to match mockup

designerneil committed:
Adds margins between paragraphs to match mockup
Adds margin to the left of the article to match mockup

designerneil committed:
Adds margins between paragraphs to match mockup
Adds margin to the left of the article to match mockup

designerneil committed:
Adds margins between paragraphs to match mockup
Adds margin to the left of the article to match mockup

designerneil committed:
Adds margins between paragraphs to match mockup
Adds margin to the left of the article to match mockup

designerneil committed:
Adds margins between paragraphs to match mockup
Adds margin to the left of the article to match mockup

designerneil committed:
Adds margins between paragraphs to match mockup
Adds margin to the left of the article to match mockup

designerneil committed:
Adds margins between paragraphs to match mockup
Adds margin to the left of the article to match mockup

designerneil committed:
Adds margins between paragraphs to match mockup
Adds margin to the left of the article to match mockup

designerneil committed:
Merge branch 'bugfix/EWL-4424-Create-News-Article-Template' of github.com:AmericanMedicalAssociation/ama-style-guide-2 into bugfix/EWL-4424-Create-News-Article-Template

designerneil commented:
@labbydev I've made changes to the two column template so it can accommodate the margin in the article body so it matches the inline bio margin in the text. In addition, the paragraph now matches the mockup a bit more closely.

labbydev commented on styleguide/source/_patterns/05-pages/news.twig:
```diff
@@ -9,6 +9,8 @@
 
 {% block pageContent %}
+  {% include '@atoms/paragraph.twig' %}
+  {% include '@atoms/paragraph.twig' %}
```
It would be more efficient to do this as a `foreach` so that we can add as many paragraphs as is necessary and we don't have to update the `news.twig` if we need to update that data in the `.json`

labbydev commented on styleguide/source/assets/scss/01-atoms/_links.scss:
```diff
@@ -67,6 +67,11 @@ a {
 
+//Adds underlines to all links inside of p tags
+p a {
+  text-decoration: underline;
```
Are all links that don't have a specific class going to need this or only items that are inside `<p>` tags? I'm concerned that this could get buggy when dealing with WYSIWYG content or other content.

designerneil replied:
only links that are inside of p tags are going have underlines.

designerneil replied:
let's revisit this when we implement the WYSIWYG  editors. It could get funky

labbydev commented on styleguide/source/_patterns/02-molecules/inline-bio.twig:
```diff
@@ -1,19 +1,21 @@
 
-  {% set heading = bio.heading %}
-  {% include "@atoms/heading.twig" %}
+  <div class="ama__inline-bio__text">
```
what was the purpose of this change? Seems irrelevant to this ticket.

designerneil replied:
this was done because the article body copy needs align with the bio text above it per UX comp.

labbydev commented on styleguide/source/_patterns/03-organisms/four-up-teaser.twig:
```diff
@@ -1,6 +1,6 @@
 <div class="ama__four-up-teaser">
-  {% for item in fourUpTeaser.items[0:4] %}
```
This shouldn't be removed. It's an important note that the items never exceed 4 even when more than 4 are added.

labbydev commented on styleguide/source/assets/scss/00-base/_typography.scss:
```diff
@@ -44,6 +44,12 @@ h5,
   @include font-size($h5-font-sizes);
 }
 
+p {
```
Should these styles be on the `body`?

designerneil replied:
body already has a style but the p tag had different sizing. This is to ensure all p are the same size no matter what.

labbydev replied:
I agree that all `<p>` should be the same I guess my question is why is the body not set to the default `<p>` sizing?

labbydev replied:
If we need to change the size of the body copy of page that needs to happen on the default html element. All sizing of everything else should be relative to the base body copy. This kind of deviation is going to be a nightmare for maintaining the code moving forward.

designerneil replied:
agreed making the appropriate changes now and removing the a tag inside of p.

labbydev commented on styleguide/source/assets/scss/01-atoms/_icons.scss:
```diff
@@ -43,7 +43,7 @@ $icons: (
   .icon--#{$name} {
     display: inline-block;
-    background: url("../../assets/icons/svg/icon-" + $name + ".svg") no-repeat 0 0;
+    background: url("../icons/svg/icon-" + $name + ".svg") no-repeat 0 0;
```
All of the icons are no longer appearing now. This pulls in a broken link.

designerneil replied:
hmmm. this was done so in D8 the icon paths were not broken. The icons are not broken on my machine. Perhaps we can compare notes?

designerneil replied:
Did you try the gulp task gulp svg2twig?

labbydev commented on styleguide/source/assets/scss/02-molecules/_tags.scss:
```diff
@@ -1,16 +1,23 @@
-  padding: ($gutter / 4) ($gutter / 2);
-  background: $gray-7;
-  color: $purple;
+  .ama__tag {
```
This should probably not be nested within `ama__tags`

labbydev reviewed (commented):
Please remove any irrelevant changes, you should be able to use `git checkout SHA -- path/to/filename` to remove any files that were changed unnecessarily.

designerneil committed:
changes names of article-metadata to better reflect BEM standards for what the molecule is

designerneil committed:
Merge branch 'chore/EWL-4455-remove-old-typography-colors' of github.com:AmericanMedicalAssociation/ama-style-guide-2 into chore/EWL-4455-remove-old-typography-colors

Commit:
Merge pull request #59 from AmericanMedicalAssociation/feature/EWL-4388-migrate-lead-news-block

[EWL-4388] Migrate Lead News Block

Commit:
Merge branch 'develop' into chore/EWL-4455-remove-old-typography-colors

Commit:
Merge pull request #64 from AmericanMedicalAssociation/chore/EWL-4455-remove-old-typography-colors

EWL-4455 removes old colors and fonts

designerneil committed:
Takes ama__tag out of ama__tags

designerneil committed:
Takes ama__tag out of ama__tags

designerneil committed:
Takes ama__tag out of ama__tags

designerneil committed:
Takes ama__tag out of ama__tags

designerneil committed:
Merge branch 'bugfix/EWL-4424-Create-News-Article-Template' of github.com:AmericanMedicalAssociation/ama-style-guide-2 into bugfix/EWL-4424-Create-News-Article-Template

designerneil committed:
Merge branch 'bugfix/EWL-4424-Create-News-Article-Template' of github.com:AmericanMedicalAssociation/ama-style-guide-2 into bugfix/EWL-4424-Create-News-Article-Template

designerneil committed:
Adds back array restrictor for fourUpTeaser

designerneil committed:
Adds back array restrictor for fourUpTeaser

designerneil committed:
Adds back array restrictor for fourUpTeaser

designerneil committed:
Merge branch 'bugfix/EWL-4424-Create-News-Article-Template' of github.com:AmericanMedicalAssociation/ama-style-guide-2 into bugfix/EWL-4424-Create-News-Article-Template

designerneil committed:
Merge branch 'bugfix/EWL-4418-social-share' of github.com:AmericanMedicalAssociation/ama-style-guide-2 into bugfix/EWL-4418-social-share

designerneil committed:
Adds missing icon files

designerneil committed:
Adds missing icon files

Commit:
Merge pull request #69 from AmericanMedicalAssociation/bugfix/EWL-4418-social-share

EWL-4418 Adds social share buttons to article header

designerneil committed:
update ribbon to use the ama__white-link so I don't have to put an a tag inside of a p tag

designerneil committed:
update ribbon to use the ama__white-link so I don't have to put an a tag inside of a p tag

designerneil committed:
Merge branch 'bugfix/EWL-4424-Create-News-Article-Template' of github.com:AmericanMedicalAssociation/ama-style-guide-2 into bugfix/EWL-4424-Create-News-Article-Template

designerneil committed:
Updates the way links behave.
By default links have and underline
Links with classes of ama__link--blue, ama__link--purple etc have no underlines
Removes class p tag with an a tag inside of it
